File: src/eqn/ast/EqnAstNodeCustomUnaryFunction.java
package eqn.ast;

import eqn.parser.exception.EqnException;
import org.jetbrains.annotations.NotNull;

public class EqnAstNodeCustomUnaryFunction extends EqnAstNodeUnaryFunction {

    public EqnAstNodeCustomUnaryFunction(String value, @NotNull EqnAstNode equationNode) throws EqnException {
        super(value, Type.CustomUnaryFunction, equationNode, PrecedenceType.Function);
    }

    @Override
    public String toString() {
        return value + '(' + operand().toString() + ')';
    }

    @Override
    public double evaluate() throws EqnException {
        throw new EqnException("Unary custom function evaluate not yet implemented (make this class abstract)!");
    }

    @Override
    public EqnAstNode simplify() throws EqnException {
        simplifyChildren();
        return this;
    }
}

File: src/eqn/ast/EqnAstNodeCustomUnaryFunction.kt
package eqn.ast

import eqn.ast.base.EqnAstNode
import eqn.ast.base.EqnAstNodeUnary
import eqn.parser.exception.EqnException

class EqnAstNodeCustomUnaryFunction(value: String, equationNode: EqnAstNode) : EqnAstNodeUnary(value, Type.CustomUnaryFunction, PrecedenceType.Function, equationNode) {
    override fun toString(): String {
        return "$value($operand)"
    }

    @Throws(EqnException::class)
    override fun evaluate(arguments: Map<String, Double>?): Double {
        throw EqnException("Unary custom function evaluate not yet implemented (make this class abstract)!")
    }

    @Throws(EqnException::class)
    override fun simplify(arguments: Map<String, Double>?): EqnAstNode {
        simplifyChildren(arguments)
        return this
    }
}